packages/tools/src/tools/annotation/PlanarFreehandROITool.ts

```diff
@@ -578,6 +576,20 @@ class PlanarFreehandROITool extends AnnotationTool {
     return annotationsWithinSlice;
   }
 
+  protected getAnnotationStyle(context) {
+    // This method exists only because `super` cannot be called from
+    // _getRenderingOptions() which is in an external file.
+    return super.getAnnotationStyle(context);
+  }
+
+  protected getPolylinePoints(annotation: Annotation): Types.Point2[] {
+    // Combine code from renderContourBeingDrawn, renderClosedContourBeingEdited
+    // and renderOpenContourBeingEdited to return the polyline points that needs
+    // for the segments that need to be rendered. Currently they are being
+    // rendered by this class but it should be rendered by the base class.
+    return [];
+  }
```

Comment:
If it relies only on `annotation`, then this is a utility not a class member

Reply:
Maybe I'm wrong

Reply:
That's because annotation data objects may store contour data (polyline) on different places then each child class need to override and return the polyline points. A way to remove this method is making all contour annotations data object similar (extend).

Reply:
Ok